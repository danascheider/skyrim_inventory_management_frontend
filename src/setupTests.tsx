import { type ReactElement } from 'react'
import { BrowserRouter } from 'react-router-dom'
import { render } from '@testing-library/react'

export const renderWithRouter = (ui: ReactElement) => render(<BrowserRouter>{ui}</BrowserRouter>)