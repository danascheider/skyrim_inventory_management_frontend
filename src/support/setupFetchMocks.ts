import { vi } from 'vitest'
import createFetchMock from 'vitest-fetch-mock'

/*
 *
 * API Mock Setup
 *
 */

const fetchMocker = createFetchMock(vi)

fetchMocker.enableMocks()
