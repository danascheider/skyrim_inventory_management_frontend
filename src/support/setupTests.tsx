import { type ReactElement } from 'react'
import { BrowserRouter } from 'react-router-dom'
import { render } from '@testing-library/react'
import { LoginContext } from '../contexts/loginContext'
import { User } from 'firebase/auth'
import testProfileImg from './testProfileImg.png'

export const requireLogin = () => { /* noop */ }

export const testUser = {
  uid: 'edna',
  displayName: 'Edna St. Vincent Millay',
  email: 'edna@gmail.com',
  photoURL: testProfileImg,
  emailVerified: true,
  getIdToken: () => new Promise<string>((resolve, _reject) => resolve('xxxxxxx'))
} as User

export const renderWithRouter = (ui: ReactElement) =>
  render(<BrowserRouter>{ui}</BrowserRouter>)

export const renderAuthenticated = (
  ui: ReactElement,
  authLoading = false,
  authError = undefined
) =>
  renderWithRouter(
    <LoginContext.Provider value={{ user: testUser, token: 'xxxxxxx', requireLogin, authLoading, authError }}>
      {ui}
    </LoginContext.Provider>
  )

export const renderUnauthenticated = (ui: ReactElement) =>
  renderWithRouter(
    <LoginContext.Provider value={{ user: null, token: null, authLoading: false, requireLogin }}>
      {ui}
    </LoginContext.Provider>
  )

export const renderAuthLoading = (ui: ReactElement) =>
  renderWithRouter(
    <LoginContext.Provider value={{ user: null, token: null, authLoading: true, requireLogin }}>
      {ui}
    </LoginContext.Provider>
  )
