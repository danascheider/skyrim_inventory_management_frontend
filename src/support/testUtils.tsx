import { type ReactElement } from 'react'
import { BrowserRouter } from 'react-router-dom'
import { JSDOM } from 'jsdom'
import { render } from '@testing-library/react'
import { LoginContext } from '../contexts/loginContext'
import { User } from 'firebase/auth'
import testProfileImg from './testProfileImg.png'

export const BASE_APP_URI = 'http://localhost:5173'

declare global {
  namespace NodeJS {
    interface Global {
      document: Document
      window: Window
    }
  }
}

const setDom = () => {
  const dom = new JSDOM('<!doctype html><html><body></body></html>', {
    url: 'http://localhost:5173',
  })

  global.window = dom.window as unknown as Window & typeof globalThis
  global.document = dom.window.document
}

/*
 *
 * Global Test Data
 *
 */

export const requireLogin = () => {
  /* noop */
}

export const testUser = {
  uid: 'edna',
  displayName: 'Edna St. Vincent Millay',
  email: 'edna@gmail.com',
  photoURL: testProfileImg,
  emailVerified: true,
  getIdtoken: () =>
    new Promise<string>((resolve, _reject) => resolve('xxxxxxx')),
} as unknown as User

/*
 *
 * Test Renderers
 *
 */

const renderWithJSDOM = (ui: ReactElement) => {
  setDom()

  return render(ui)
}

export const renderWithRouter = (ui: ReactElement) =>
  renderWithJSDOM(<BrowserRouter>{ui}</BrowserRouter>)

export const renderAuthenticated = (ui: ReactElement, authLoading = false) =>
  renderWithRouter(
    <LoginContext.Provider
      value={{
        user: testUser,
        token: 'xxxxxxx',
        requireLogin,
        authLoading,
      }}
    >
      {ui}
    </LoginContext.Provider>
  )

export const renderUnauthenticated = (ui: ReactElement) =>
  renderWithRouter(
    <LoginContext.Provider
      value={{
        user: null,
        token: null,
        authLoading: false,
        requireLogin,
      }}
    >
      {ui}
    </LoginContext.Provider>
  )

export const renderAuthLoading = (ui: ReactElement) =>
  renderWithRouter(
    <LoginContext.Provider
      value={{ user: null, token: null, authLoading: true, requireLogin }}
    >
      {ui}
    </LoginContext.Provider>
  )
