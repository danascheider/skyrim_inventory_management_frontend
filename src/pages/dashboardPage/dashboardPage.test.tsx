import { describe, test, expect } from 'vitest'
import { screen } from '@testing-library/react'
import { renderAuthenticated } from '../../support/setupTests'
import DashboardPage from './dashboardPage'

describe('<DashboardPage />', () => {
  test('DashboardPage displays the header', () => {
    const wrapper = renderAuthenticated(<DashboardPage />)
    expect(wrapper).toBeTruthy()

    const h1 = wrapper.container.querySelector('h1')
    // This is because 'Skyrim Inventory 'Management' and 'S. I. M.' are
    // both in the DOM but visible at different viewport sizes. Since
    // vitest and testing-library don't give us the ability to set a viewport
    // size for tests, both will show up here.
    expect(h1?.textContent).toBe('Skyrim Inventory ManagementS. I. M.')
  })

  test('DashboardPage displays user info', () => {
    const wrapper = renderAuthenticated(<DashboardPage />)

    expect(screen.getByText('Edna St. Vincent Millay')).toBeTruthy()
    expect(screen.getByText('edna@gmail.com')).toBeTruthy()

    const img = wrapper.container.querySelector('img')
    expect(img).toBeTruthy()
    expect(img?.src).toEqual('/src/testProfileImg.png')
  })

  test('DashboardPage displays the navigation mosaic', () => {
    renderAuthenticated(<DashboardPage />)

    expect(screen.getByText('Your Games')).toBeTruthy()
    expect(screen.getByText('Your Shopping Lists')).toBeTruthy()
    expect(screen.getByText('Your Inventory')).toBeTruthy()
    expect(screen.getByText('Nav Link 4')).toBeTruthy()
    expect(screen.getByText('Nav Link 5')).toBeTruthy()
  })

  test('matches snapshot', () => {
    const wrapper = renderAuthenticated(<DashboardPage />)
    expect(wrapper).toMatchSnapshot()
  })
})
