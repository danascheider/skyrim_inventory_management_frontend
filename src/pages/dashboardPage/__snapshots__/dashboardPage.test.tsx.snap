// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<DashboardPage /> > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="_root_03d657"
      >
        <section
          class="_container_03d657"
        >
          <div
            class="_root_30a763"
          >
            <div
              class="_root_dcf87f"
            >
              <div
                class="_card_dcf87f"
              >
                <a
                  class="_root_396ea1"
                  href="/games"
                  style="--background-color: #ffbf00; --hover-color: #e5ab00; --text-color: #4c3900;"
                >
                  Your Games
                </a>
              </div>
              <div
                class="_card_dcf87f"
              >
                <a
                  class="_root_396ea1"
                  href="/"
                  style="--background-color: #e83f6f; --hover-color: #d03863; --text-color: #fff;"
                >
                  Your Shopping Lists
                </a>
              </div>
              <div
                class="_card_dcf87f"
              >
                <a
                  class="_root_396ea1"
                  href="/"
                  style="--background-color: #2274a5; --hover-color: #1e6894; --text-color: #fff;"
                >
                  Your Inventory
                </a>
              </div>
              <div
                class="_card_dcf87f"
              >
                <a
                  class="_root_396ea1"
                  href="/"
                  style="--background-color: #00a323; --hover-color: #00921f; --text-color: #fff;"
                >
                  Nav Link 4
                </a>
              </div>
              <div
                class="_card_dcf87f"
              >
                <a
                  class="_root_396ea1"
                  href="/"
                  style="--background-color: #20e2e9; --hover-color: #11cbd1; --text-color: #094345;"
                >
                  Nav Link 5
                </a>
              </div>
            </div>
          </div>
        </section>
        <header
          class="_root_c29812"
        >
          <div
            class="_bar_c29812"
          >
            <span
              class="_headerContainer_c29812"
            >
              <h1
                class="_header_c29812"
              >
                <a
                  class="_headerLinkLarge_c29812"
                  href="/dashboard"
                >
                  Skyrim Inventory
                  <br
                    class="_bp_c29812"
                  />
                   Management
                </a>
                <a
                  class="_headerLinkSmall_c29812"
                  href="/dashboard"
                >
                  S. I. M.
                </a>
              </h1>
            </span>
            <span
              class="_root_d263fb"
            >
              <div
                class="_main_d263fb"
              >
                <div
                  class="_button_d263fb"
                  role="button"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-bars _hamburger_d263fb"
                    data-icon="bars"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
                      fill="currentColor"
                    />
                  </svg>
                  <span
                    class="_info_d263fb"
                  >
                    <p
                      class="_name_d263fb"
                    >
                      Edna St. Vincent Millay
                    </p>
                    <p
                      class="_email_d263fb"
                    >
                      edna@gmail.com
                    </p>
                  </span>
                  <img
                    alt="User profile image"
                    class="_img_d263fb"
                    referrerpolicy="no-referrer"
                    src="/src/support/testProfileImg.png"
                  />
                </div>
              </div>
              <menu
                class="_dropdown_d263fb"
              >
                <div>
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-right-from-bracket "
                    data-icon="right-from-bracket"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M377.9 105.9L500.7 228.7c7.2 7.2 11.3 17.1 11.3 27.3s-4.1 20.1-11.3 27.3L377.9 406.1c-6.4 6.4-15 9.9-24 9.9c-18.7 0-33.9-15.2-33.9-33.9l0-62.1-128 0c-17.7 0-32-14.3-32-32l0-64c0-17.7 14.3-32 32-32l128 0 0-62.1c0-18.7 15.2-33.9 33.9-33.9c9 0 17.6 3.6 24 9.9zM160 96L96 96c-17.7 0-32 14.3-32 32l0 256c0 17.7 14.3 32 32 32l64 0c17.7 0 32 14.3 32 32s-14.3 32-32 32l-64 0c-53 0-96-43-96-96L0 128C0 75 43 32 96 32l64 0c17.7 0 32 14.3 32 32s-14.3 32-32 32z"
                      fill="currentColor"
                    />
                  </svg>
                  <p
                    class="_signOutText_d263fb"
                  >
                    Sign Out
                  </p>
                </div>
              </menu>
            </span>
          </div>
        </header>
        <div
          class="_root_91d025 _hidden_91d025"
          style="--text-color: #4e6d83;"
        />
        <div
          class="_root_753dd0 _hidden_753dd0"
          data-testid="modal"
        >
          <div
            class="_content_753dd0"
          />
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="_root_03d657"
    >
      <section
        class="_container_03d657"
      >
        <div
          class="_root_30a763"
        >
          <div
            class="_root_dcf87f"
          >
            <div
              class="_card_dcf87f"
            >
              <a
                class="_root_396ea1"
                href="/games"
                style="--background-color: #ffbf00; --hover-color: #e5ab00; --text-color: #4c3900;"
              >
                Your Games
              </a>
            </div>
            <div
              class="_card_dcf87f"
            >
              <a
                class="_root_396ea1"
                href="/"
                style="--background-color: #e83f6f; --hover-color: #d03863; --text-color: #fff;"
              >
                Your Shopping Lists
              </a>
            </div>
            <div
              class="_card_dcf87f"
            >
              <a
                class="_root_396ea1"
                href="/"
                style="--background-color: #2274a5; --hover-color: #1e6894; --text-color: #fff;"
              >
                Your Inventory
              </a>
            </div>
            <div
              class="_card_dcf87f"
            >
              <a
                class="_root_396ea1"
                href="/"
                style="--background-color: #00a323; --hover-color: #00921f; --text-color: #fff;"
              >
                Nav Link 4
              </a>
            </div>
            <div
              class="_card_dcf87f"
            >
              <a
                class="_root_396ea1"
                href="/"
                style="--background-color: #20e2e9; --hover-color: #11cbd1; --text-color: #094345;"
              >
                Nav Link 5
              </a>
            </div>
          </div>
        </div>
      </section>
      <header
        class="_root_c29812"
      >
        <div
          class="_bar_c29812"
        >
          <span
            class="_headerContainer_c29812"
          >
            <h1
              class="_header_c29812"
            >
              <a
                class="_headerLinkLarge_c29812"
                href="/dashboard"
              >
                Skyrim Inventory
                <br
                  class="_bp_c29812"
                />
                 Management
              </a>
              <a
                class="_headerLinkSmall_c29812"
                href="/dashboard"
              >
                S. I. M.
              </a>
            </h1>
          </span>
          <span
            class="_root_d263fb"
          >
            <div
              class="_main_d263fb"
            >
              <div
                class="_button_d263fb"
                role="button"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-bars _hamburger_d263fb"
                  data-icon="bars"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
                    fill="currentColor"
                  />
                </svg>
                <span
                  class="_info_d263fb"
                >
                  <p
                    class="_name_d263fb"
                  >
                    Edna St. Vincent Millay
                  </p>
                  <p
                    class="_email_d263fb"
                  >
                    edna@gmail.com
                  </p>
                </span>
                <img
                  alt="User profile image"
                  class="_img_d263fb"
                  referrerpolicy="no-referrer"
                  src="/src/support/testProfileImg.png"
                />
              </div>
            </div>
            <menu
              class="_dropdown_d263fb"
            >
              <div>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-right-from-bracket "
                  data-icon="right-from-bracket"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M377.9 105.9L500.7 228.7c7.2 7.2 11.3 17.1 11.3 27.3s-4.1 20.1-11.3 27.3L377.9 406.1c-6.4 6.4-15 9.9-24 9.9c-18.7 0-33.9-15.2-33.9-33.9l0-62.1-128 0c-17.7 0-32-14.3-32-32l0-64c0-17.7 14.3-32 32-32l128 0 0-62.1c0-18.7 15.2-33.9 33.9-33.9c9 0 17.6 3.6 24 9.9zM160 96L96 96c-17.7 0-32 14.3-32 32l0 256c0 17.7 14.3 32 32 32l64 0c17.7 0 32 14.3 32 32s-14.3 32-32 32l-64 0c-53 0-96-43-96-96L0 128C0 75 43 32 96 32l64 0c17.7 0 32 14.3 32 32s-14.3 32-32 32z"
                    fill="currentColor"
                  />
                </svg>
                <p
                  class="_signOutText_d263fb"
                >
                  Sign Out
                </p>
              </div>
            </menu>
          </span>
        </div>
      </header>
      <div
        class="_root_91d025 _hidden_91d025"
        style="--text-color: #4e6d83;"
      />
      <div
        class="_root_753dd0 _hidden_753dd0"
        data-testid="modal"
      >
        <div
          class="_content_753dd0"
        />
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<DashboardPage /> > when auth is loading > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="_root_03d657"
      >
        <section
          class="_container_03d657"
        >
          <div
            class="_root_30a763"
          >
            <span
              data-testid="pulseLoader"
              style="display: inherit; text-align: center;"
            >
              <span
                style="background-color: rgb(255, 203, 50); width: 15px; height: 15px; margin: 2px; border-radius: 100%; display: inline-block; animation: react-spinners-PulseLoader-pulse 0.75s 0.12s infinite cubic-bezier(0.2, 0.68, 0.18, 1.08); animation-fill-mode: both;"
              />
              <span
                style="background-color: rgb(255, 203, 50); width: 15px; height: 15px; margin: 2px; border-radius: 100%; display: inline-block; animation: react-spinners-PulseLoader-pulse 0.75s 0.24s infinite cubic-bezier(0.2, 0.68, 0.18, 1.08); animation-fill-mode: both;"
              />
              <span
                style="background-color: rgb(255, 203, 50); width: 15px; height: 15px; margin: 2px; border-radius: 100%; display: inline-block; animation: react-spinners-PulseLoader-pulse 0.75s 0.36s infinite cubic-bezier(0.2, 0.68, 0.18, 1.08); animation-fill-mode: both;"
              />
            </span>
          </div>
        </section>
        <header
          class="_root_c29812"
        >
          <div
            class="_bar_c29812"
          >
            <span
              class="_headerContainer_c29812"
            >
              <h1
                class="_header_c29812"
              >
                <a
                  class="_headerLinkLarge_c29812"
                  href="/dashboard"
                >
                  Skyrim Inventory
                  <br
                    class="_bp_c29812"
                  />
                   Management
                </a>
                <a
                  class="_headerLinkSmall_c29812"
                  href="/dashboard"
                >
                  S. I. M.
                </a>
              </h1>
            </span>
            <div
              class="_main_d263fb"
            >
              <div
                class="_button_d263fb"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-bars _hamburger_d263fb"
                  data-icon="bars"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
                    fill="currentColor"
                  />
                </svg>
                <img
                  alt="User profile image"
                  class="_img_d263fb"
                  src="/src/components/userInfo/anonymousAvatar.jpg"
                />
              </div>
            </div>
          </div>
        </header>
        <div
          class="_root_91d025 _hidden_91d025"
          style="--text-color: #4e6d83;"
        />
        <div
          class="_root_753dd0 _hidden_753dd0"
          data-testid="modal"
        >
          <div
            class="_content_753dd0"
          />
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="_root_03d657"
    >
      <section
        class="_container_03d657"
      >
        <div
          class="_root_30a763"
        >
          <span
            data-testid="pulseLoader"
            style="display: inherit; text-align: center;"
          >
            <span
              style="background-color: rgb(255, 203, 50); width: 15px; height: 15px; margin: 2px; border-radius: 100%; display: inline-block; animation: react-spinners-PulseLoader-pulse 0.75s 0.12s infinite cubic-bezier(0.2, 0.68, 0.18, 1.08); animation-fill-mode: both;"
            />
            <span
              style="background-color: rgb(255, 203, 50); width: 15px; height: 15px; margin: 2px; border-radius: 100%; display: inline-block; animation: react-spinners-PulseLoader-pulse 0.75s 0.24s infinite cubic-bezier(0.2, 0.68, 0.18, 1.08); animation-fill-mode: both;"
            />
            <span
              style="background-color: rgb(255, 203, 50); width: 15px; height: 15px; margin: 2px; border-radius: 100%; display: inline-block; animation: react-spinners-PulseLoader-pulse 0.75s 0.36s infinite cubic-bezier(0.2, 0.68, 0.18, 1.08); animation-fill-mode: both;"
            />
          </span>
        </div>
      </section>
      <header
        class="_root_c29812"
      >
        <div
          class="_bar_c29812"
        >
          <span
            class="_headerContainer_c29812"
          >
            <h1
              class="_header_c29812"
            >
              <a
                class="_headerLinkLarge_c29812"
                href="/dashboard"
              >
                Skyrim Inventory
                <br
                  class="_bp_c29812"
                />
                 Management
              </a>
              <a
                class="_headerLinkSmall_c29812"
                href="/dashboard"
              >
                S. I. M.
              </a>
            </h1>
          </span>
          <div
            class="_main_d263fb"
          >
            <div
              class="_button_d263fb"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-bars _hamburger_d263fb"
                data-icon="bars"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
                  fill="currentColor"
                />
              </svg>
              <img
                alt="User profile image"
                class="_img_d263fb"
                src="/src/components/userInfo/anonymousAvatar.jpg"
              />
            </div>
          </div>
        </div>
      </header>
      <div
        class="_root_91d025 _hidden_91d025"
        style="--text-color: #4e6d83;"
      />
      <div
        class="_root_753dd0 _hidden_753dd0"
        data-testid="modal"
      >
        <div
          class="_content_753dd0"
        />
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
